recipes-support/host-insight-helper/host-insight-helper/host-insight-helper.sh

```diff
@@ -1,17 +1,40 @@
 #!/bin/sh
 
 ID_PATH="/etc/opt/host-insight-client/identity-fallback.toml"
+CLIENT_UPGRADE="/tmp/host-insight/client_upgrade"
 
-# Get the serial number of the device.
-if [ "$(cat /etc/platform-system-type)" = "c61" ]; then
-  # Sleep until we know that pic_upgrade is complete
-  sleep 30
-  echo "uid = \"$(cat /opt/hm/pic_attributes/ctrl_serial_nr)\"" > $ID_PATH
-else
-  echo "uid = \"$(cat /proc/device-tree/chosen/serial-number)\"" > $ID_PATH
-fi
+upgrade_client()
+{
+  systemctl stop host-insight-client
+  echo "Upgrading host-insight-client to version $new..."
+  read -r new < "$CLIENT_UPGRADE"
+  opkg remove "host-insight-client"
+  if ! opkg install "host-insight-client$new"; then
+    echo "Failed to upgrade to version $new. Falling back to installing 'host-insight-client'"
+    opkg install "host-insight-client"
+  fi
+  rm "$CLIENT_UPGRADE"
+  systemctl start host-insight-client
+}
+
+assign_uid_and_fallback_domain()
+{
+  # Get the serial number of the device.
+  if [ "$(cat /etc/platform-system-type)" = "c61" ]; then
+    # Sleep until we know that pic_upgrade is complete
+    sleep 30
```

Comment:
You can check if the system is running pic_upgrade.sh

Reply:
This is existing code that I just moved to a separate function. What do you suggest changing?

Reply:
By the way, is pgrep included in the Host Insight BSB? I made it work with out without it, but we might as well have it I think.

Reply:
Not sure I have not used it, but the fall back should work too